src/modules/Api.ts

```diff
@@ -183,6 +192,159 @@ export class Api {
     });
   }
 
+  public async canonicalOrder({
+    order: {
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration = new BigNumber(FOUR_WEEKS_IN_SECONDS),
+      limitFee,
+    },
+    fillOrKill,
+    postOnly,
+    clientId,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
+    order: {
+      side: ApiSide,
+      market: ApiMarketName,
+      amount: BigNumberable,
+      price: BigNumberable,
+      makerAccountOwner: address,
+      expiration: BigNumberable,
+      limitFee?: BigNumberable,
+    },
+    fillOrKill?: boolean,
+    postOnly?: boolean,
+    clientId?: string,
+    cancelId?: string,
+    cancelAmountOnRevert?: boolean,
+  }): Promise<{ order: ApiOrder }> {
+    if (!side || !market) {
+      return;
+    }
```

Comment:
can you validate that they are a legit value in the ENUM?

Reply:
actually do this in `createCanonicalOrder` too so that it catches it in all cases